### PR TITLE
Expand cusp correction cases handled

### PR DESCRIPTION
**Pull request by markdewing**
Handle the case where the number of orbitals is greater than the number of electrons.
This is important for multideterminants and for orbital rotation.

For testing, an existing test case was modified.  The input wavefunction was changed to use the new ``sposet_collection`` style.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Bugfix

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Ye. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by markdewing** [comment]
Test this please

**Comment by ye-luo** [comment]
Can we expand an SPOSet level unit test to incorporate this? That should help us to catch issues even faster.
I was expecting evaluateVGL also gets affected but it doesn't seem so.

**Comment by markdewing** [comment]
evaluateVGL doesn't seem to be called.

File: tests/molecules/C4_ae/SD-C4_AE_Mol_QP.wfnoj.xml [hunk 1]
```diff
@@ -1,16 +1,16 @@
 <?xml version="1.0"?>
 <qmcsystem>
   <wavefunction name="psi0" target="e">
-    <determinantset type="MolecularOrbital" name="LCAOBSet" source="ion0" transform="yes" cuspCorrection="yes" href="C4_AE_Mol_QP.h5">
+    <sposet_collection type="MolecularOrbital" name="LCAOBSet" source="ion0" transform="yes" cuspCorrection="yes" href="C4_AE_Mol_QP.h5">
+      <sposet name="spo-up" size="92" cuspInfo="spo-up.cuspInfo.xml">
```
Comment (ye-luo) [inline tests/molecules/C4_ae/SD-C4_AE_Mol_QP.wfnoj.xml:5]: Can we keep this 12 as it was and also keep using the pre-computed cuspInfo xml file?

Comment (markdewing) [reply tests/molecules/C4_ae/SD-C4_AE_Mol_QP.wfnoj.xml:5]: I'll revert this change and add a new test

File: src/QMCWaveFunctions/tests/hcn.wfnoj.xml [hunk 2]
```diff
@@ -412,7 +412,7 @@
  -1.45805000000000e+00  0.00000000000000e+00  0.00000000000000e+00
 </coefficient>
         </determinant>
-        <determinant id="downdet" size="7" cuspInfo="hcn_downdet.cuspInfo.xml">
+        <determinant id="downdet" size="9" cuspInfo="hcn_downdet.cuspInfo.xml">
```
[the remark this answers is not in the document]
Comment (markdewing) [reply src/QMCWaveFunctions/tests/hcn.wfnoj.xml:415]: The ethanol test case still tests the case where the number of orbitals is the same as the number of electrons.

File: tests/molecules/C4_ae/SD-C4_AE_Mol_large_QP.wfnoj.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<qmcsystem>
+  <wavefunction name="psi0" target="e">
+    <sposet_collection type="MolecularOrbital" name="LCAOBSet" source="ion0" transform="yes" cuspCorrection="yes" href="C4_AE_Mol_QP.h5">
+      <!-- The is the case where the number of orbitals is greater than the number of electrons -->
+      <sposet name="spo-up" size="92" cuspInfo="spo-up.cuspInfo.xml">
```
[the remark this answers is not in the document]
Comment (ye-luo) [reply tests/molecules/C4_ae/SD-C4_AE_Mol_large_QP.wfnoj.xml:6]: That is good. `spo-up.cuspInfo.xml` has 92 and `updet.cuspInfo.xml` has 12.

**Review by ye-luo — APPROVED** [review]

**Review by ye-luo — APPROVED** [review]
Still need non-ANL approval

**Comment by ye-luo** [comment]
Test this please

**Review by PDoakORNL — APPROVED** [review]
Also looks good to me.

**Pull request merged**